src/Illuminate/Validation/Rules/Uuid.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Illuminate\Validation\Rules;
+
+use Stringable;
+
+class Uuid implements Stringable
+{
+    protected ?int $version = null;
```

Comment:
`Str::isUuid()` actually also accepts `"nil"` and `"max"`
```suggestion
    protected int|'nil'|'max'|null $version = null;
```

Reply:
But the docs say this:

> [uuid](https://laravel.com/docs/12.x/validation#rule-uuid)
> The field under validation must be a valid RFC 9562 (version 1, 3, 4, 5, 6, 7, or 8) universally unique identifier (UUID).
> You may also validate that the given UUID matches a UUID specification by version: 
'uuid' => 'uuid:4'

So I would understand that is different than the proposed `'nil'` and `'max'` values?

For this case, I didn’t add version validation like I did in the `IpAddress` rule because IP formats are less likely to change, but UUID versions might evolve. However, we could definitely add version validation for UUIDs as well if needed.

Reply:
But the docs say this:

> [uuid](https://laravel.com/docs/12.x/validation#rule-uuid)
> The field under validation must be a valid RFC 9562 (version 1, 3, 4, 5, 6, 7, or 8) universally unique identifier (UUID).
> You may also validate that the given UUID matches a UUID specification by version: 
'uuid' => 'uuid:4'

So I would understand that is different than the proposed `'nil'` and `'max'` values?

Reply:
A nil UUID is `00000000-0000-0000-0000-000000000000` (the only supported UUID for when the version position is set to `0`, while a max UUID is where that position as well as every other position in the UUID is set the highest value: `FFFFFFFF-FFFF-FFFF-FFFF-FFFFFFFFFFFF`

Yeah, I'd go with a docs update then.

https://github.com/laravel/framework/blob/fc4cef42f1ea6b724e6fda7853b86d78b2d5d45d/src/Illuminate/Support/Str.php#L643-L676